spec/models/programs_spec.rb

```diff
@@ -9,4 +9,15 @@
   subject { FactoryGirl.create(:program) }
   it { should validate_uniqueness_of(:title) }
   it { should validate_uniqueness_of(:url) }
+
+end
+
+describe Program, '#average' do
+  it 'returns the average of the users ratings' do
+    program = FactoryGirl.create(:program)
+    FactoryGirl.create(:review, program: program, rating: 2)
+    FactoryGirl.create(:review, program: program, rating: 5)
+
+    expect(program.average).to eq 3.5
+  end
```

Comment:
I would add a test to cover the case of `0` submissions, since you were obviously thinking about that. I think you'll find your existing code works even without the `if...else`